### PR TITLE
Refactor cache configuration settings

### PR DESCRIPTION
**Pull request by ptschip**
Use a struct to store the config settings which can then
be passed around.

add tests

### CODE REVIEW

**Review by sickpig — APPROVED** [review]

File: src/txdb.cpp [hunk 1]
```diff
@@ -697,12 +697,7 @@ uint64_t GetTotalSystemMemory()
 }
 #endif
 
-void GetCacheConfiguration(int64_t &_nBlockDBCache,
-    int64_t &_nBlockUndoDBcache,
-    int64_t &_nBlockTreeDBCache,
-    int64_t &_nCoinDBCache,
-    int64_t &_nTxIndexCache,
-    bool fDefault)
+CacheConfig SetCacheConfiguration(bool fDefault)
```
Comment (gandrewstone) [inline src/txdb.cpp]: Switching from "Get" to "Set" here is showing how this naming is not topically right.  How about "Calculate", "Derive", or "Discover" -- e.g. "DiscoverCacheConfiguration"?

Also, please add doc comment

File: src/txdb.h [hunk 2]
```diff
@@ -53,24 +53,28 @@ static const int64_t nMaxBlockDBCache = 2;
 uint64_t GetAvailableMemory();
 /** Get the total physical memory */
 uint64_t GetTotalSystemMemory();
+
+struct CacheConfig
```
Comment (gandrewstone) [inline src/txdb.h:65]: Please add a doccomment for this structure and each member in it.

I'm not asking (yet) that we exhaustively go though and doc-comment everything, but I'd like to see us do so for everything we touch when doing other work.

**Review by gandrewstone — CHANGES_REQUESTED** [review]

File: src/txdb.cpp (hunk 1 above)
Comment (ptschip) [reply src/txdb.cpp]: done

File: src/txdb.h (hunk 2 above)
Comment (ptschip) [reply src/txdb.h:65]: done

**Pull request merged**